web-masons/pyramid-zappa-api-boilerplate
# Updated Warcry Demo Project

Updated Demo Warcry API Project
Connects to a MySQL AWS RDS Database.

Creates a single table called demo_users, you are able to do all the CRUD requests via an API

### Demo
Curl examples : 
#Get a list of all users in the database
curl http://<server_ip:port>/all_users

#Get a single user in database using the users id
curl http://<server_ip:port>/single_user/<users_id>

#Insert a User
curl -X POST -H "Content-Type: application/json" -d '{<columns and values>}' http://<server_ip:port>/create_new_user**
E.g : 
curl -X POST -H "Content-Type: application/json" -d '{"name": "Test User", "super_hero":"1"}' http://0.0.0.0:6543/create_new_user

#Update a User
curl -X PUT -H "Content-Type: application/json" -d '{<columns and values to be updated>}' http://<server_ip:port>/update_user/<user_id_to_update>

#Delete a User
curl -X DELETE http://<serverip:port>/delete_user/<user_id_to_delete>

### Notes
You will need to run the bootstrap script to install all required packages

## Testing Instructions

- Run the bootstrap script
- Run the setup file in the project
*python setup.py develop
- Run database initialising to create table 
* initialize_battlecry_proj_db development.ini
- Start Project to test with the above curl examples 
* pserve development.ini --reload

See Readme File in the warcry_proj folder
